src/pipeline/src/etl.rs

```diff
@@ -350,6 +367,53 @@ impl Pipeline {
     }
 }
 
+/// Transforms an array of VRL values into rows.
+fn transform_array_elements(
+    arr: &mut [VrlValue],
+    transformer: &GreptimeTransformer,
+    is_v1: bool,
+    schema_info: &mut SchemaInfo,
+    pipeline_ctx: &PipelineContext<'_>,
+) -> Result<Vec<Row>> {
+    use crate::error::{ArrayElementMustBeObjectSnafu, TransformArrayElementSnafu};
+
+    let mut rows = Vec::with_capacity(arr.len());
+
+    for (index, element) in arr.iter_mut().enumerate() {
+        if !element.is_object() {
+            return ArrayElementMustBeObjectSnafu {
+                index,
+                actual_type: element.kind_str().to_string(),
+            }
+            .fail();
```

Comment:
When users enable skip error, unexpected situations may arise here. If an element in the array is invalid, it will cause the entire array's data to fail to write. However, this can be addressed in the next pull request.

Reply:
fixed in https://github.com/GreptimeTeam/greptimedb/pull/7342/commits/f2c9ff45ca69b6cef47ec7e7856539cc57d130f8